front/app/hooks/useShowCosponsorshipReminder.ts

```diff
@@ -0,0 +1,43 @@
+import useAppConfiguration from 'api/app_configuration/useAppConfiguration';
+import useInitiativeStatus from 'api/initiative_statuses/useInitiativeStatus';
+import useInitiativeById from 'api/initiatives/useInitiativeById';
+import useAuthUser from 'api/me/useAuthUser';
+
+export default function useShowCosponsorshipReminder(initiativeId: string) {
+  const { data: appConfiguration } = useAppConfiguration();
+  const { data: initiative } = useInitiativeById(initiativeId);
+  const { data: authUser } = useAuthUser();
+  const initiativeStatusId =
+    initiative?.data.relationships.initiative_status?.data?.id;
+  const { data: initiativeStatus } = useInitiativeStatus(initiativeStatusId);
+
+  if (!initiative || !authUser || !appConfiguration || !initiativeStatus) {
+    return false;
+  }
+
+  const authorId = initiative.data.relationships.author.data?.id;
+  const signedInUserIsAuthor =
+    typeof authorId === 'string' ? authorId === authUser.data.id : false;
+  const requireReview =
+    appConfiguration.data.attributes.settings.initiatives?.require_review;
+  const requiredNumberOfCosponsors =
+    appConfiguration.data.attributes.settings.initiatives?.cosponsors_number;
+  const acceptedCosponsorships =
+    initiative.data.attributes.cosponsorships.filter(
+      (c) => c.status === 'accepted'
+    );
+  const initiativeStatusCode = initiativeStatus.data.attributes.code;
+  const initiativeHasBeenPublished =
+    initiativeStatusCode !== 'review_pending' &&
+    initiativeStatusCode !== 'changes_requested';
+
+  return (
+    signedInUserIsAuthor &&
+    typeof requiredNumberOfCosponsors === 'number' &&
+    // Showing this warning only makes sense if proposal pre-approval/review
+    // is enabled. Otherwise, proposals are published immediately anyway.
+    requireReview === true &&
```

Comment:
```suggestion
    requireReview &&
```
Is it the same?

Reply:
Not the same. === makes sure the coerced value and the type are the same. So it's a boolean that is true. Imagine if requireReview was the string "abc", it would also coerce to true but the type is string.

Reply:
But `requireReview` is `boolean | undefined`, so it cannot be `abc`. Though definitely up to you :smile:

Reply:
I know. I've changed that hook so it always returns a boolean (instead of possibly `undefined`).